ruby/ql/lib/codeql/ruby/ast/Parameter.qll

```diff
@@ -19,6 +19,27 @@ class Parameter extends AstNode, TParameter {
   /** Gets the zero-based position of this parameter. */
   final int getPosition() { this = any(Callable c).getParameter(result) }
 
+  /** Holds if this parameter is declared after a positional `*splat` parameter. */
+  private predicate isAfterSplatParameter() {
+    exists(Callable c, int n |
+      c.getParameter(n) instanceof SplatParameter and
+      this = c.getParameter([n .. c.getNumberOfParameters() - 1])
+    )
+  }
+
+  /**
+   * Gets the zero-based position of this parameter, if it is a plain positional parameter.
+   *
+   * This corresponds to the index of a positional argument that can flow into this parameter.
+   *
+   * Has no result for non-positional parameters, splat parameters, or parameters declared after a splat parameter.
```

Comment:
What about 
```
def foo (a,b=1,c)
end
```
I think in this case `c` does not have a "logical" position either.

Reply:
I see. I was under the impression that the splat parameter actually was the only thing that could offset the index of other positional parameters by a variable amount.